parser.go

```diff
@@ -138,3 +148,83 @@ func (p *Parser[V]) Pop() *Node[V] {
 	p.node = p.node.Parent
 	return n
 }
+
+// RotateLeft moves the current node to the position it's parent had,
+// making the parent into the current node's child.
+func (p *Parser[V]) RotateLeft() *Node[V] {
+	// n = node , op = original parent , gp = grand parent
+	n := p.node
+	op := n.Parent
+	gp := op.Parent
```

Comment:
What happens if this is called on the root node? I believe `op` would be `nil` and this would panic. Can we maybe return an error? Or should that be a no-op?

Reply:
Usually I prefer to detect problems and report early.  A no-op could delay detecting a problem when the calling code expected the tree to be a in state where nodes can be rotated.

What error would it return?  What if `RotateLeft()` and `AdoptSibling()` both could return the same error when required nodes to manipulate the tree are missing?
```
var ErrMissingRequiredNode = errors.New("missing required node")
```

Reply:
Pushed.